### PR TITLE
refactor(router): use ApplicationRef to provide the first bootstrapped component as ROUTER_PRIMARY_COMPONENT automatically.

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: modules/angular2/src/core/application_ref.ts [hunk 1]
```diff
@@ -313,11 +314,17 @@ export abstract class ApplicationRef {
    * Dispose of this application and all of its components.
    */
   abstract dispose(): void;
+
+  /**
+   * Get a list of component types registered to this application.
+   */
+  get componentTypes(): /*Type[]*/ any[] { return unimplemented(); };
```
Comment (IgorMinar) [inline modules/angular2/src/core/application_ref.ts]: why is the Type signature commented out?

File: modules/angular2/src/core/application_ref.ts [hunk 2]
```diff
@@ -370,4 +378,6 @@ export class ApplicationRef_ extends ApplicationRef {
     this._rootComponents.forEach((ref) => ref.dispose());
     this._platform._applicationDisposed(this);
   }
+  
+  get componentTypes(): any[] { return this._rootComponentTypes; }
```
Comment (IgorMinar) [inline modules/angular2/src/core/application_ref.ts]: @mhevery please review this part

File: modules/angular2/src/core/application_ref.ts (hunk 1 above)
Comment (alxhub) [inline modules/angular2/src/core/application_ref.ts]: No idea - it was done that way in the old bootstrap code, so I've carried it forward.

**Comment by IgorMinar** [comment]
Please add a breaking change notice into the commit message before/after examples that show people how to migrate.

otherwise looks good to me.

let's have @mhevery take a look and comment on the ApplicationRef change before merging.

File: modules/angular2/src/core/application_ref.ts (hunk 2 above)
Comment (mhevery) [inline modules/angular2/src/core/application_ref.ts]: LGTM

**Pull request closed without merging**